crates/uv-globfilter/src/portable_glob.rs

```diff
@@ -173,30 +257,64 @@ mod tests {
         );
         assert_snapshot!(
             parse_err(r"licenses\eula.txt"),
-            @r"Invalid character `\` at position 8 in glob: `licenses\eula.txt`"
+            @r"Only forward slashes are allowed as path separator, invalid character at position 8 in glob: `licenses\eula.txt`"
+        );
+        assert_snapshot!(
+            parse_err(r"**/@test"),
+            @"Invalid character `@` at position 3 in glob: `**/@test`"
+        );
+        // Escapes are not allowed in strict PEP 639 mode
+        assert_snapshot!(
+            parse_err(r"public domain/Gulliver\\’s Travels.txt"),
+            @r"Invalid character ` ` at position 6 in glob: `public domain/Gulliver\\’s Travels.txt`"
+        );
+        let parse_err_uv = |glob| {
+            let error = PortableGlobParser::Uv.parse(glob).unwrap_err();
+            anstream::adapter::strip_str(&error.to_string()).to_string()
+        };
+        assert_snapshot!(
+            parse_err_uv(r"**/@test"),
+            @"Invalid character `@` at position 3 in glob: `**/@test`. hint: Characters can be escaped with a backslash"
+        );
+        // Escaping slashes is not allowed.
+        assert_snapshot!(
+            parse_err_uv(r"licenses\\MIT.txt"),
+            @r"Path separators can't be escaped, invalid character at position 9 in glob: `licenses\\MIT.txt`"
+        );
+        assert_snapshot!(
+            parse_err_uv(r"licenses\/MIT.txt"),
+            @r"Path separators can't be escaped, invalid character at position 9 in glob: `licenses\/MIT.txt`"
         );
     }
 
     #[test]
     fn test_valid() {
         let cases = [
-            "licenses/*.txt",
-            "licenses/**/*.txt",
-            "LICEN[CS]E.txt",
-            "LICEN?E.txt",
-            "[a-z].txt",
-            "[a-z._-].txt",
-            "*/**",
-            "LICENSE..txt",
-            "LICENSE_file-1.txt",
+            r"licenses/*.txt",
+            r"licenses/**/*.txt",
+            r"LICEN[CS]E.txt",
+            r"LICEN?E.txt",
+            r"[a-z].txt",
+            r"[a-z._-].txt",
+            r"*/**",
+            r"LICENSE..txt",
+            r"LICENSE_file-1.txt",
             // (google translate)
-            "licenses/라이센스*.txt",
-            "licenses/ライセンス*.txt",
-            "licenses/执照*.txt",
-            "src/**",
+            r"licenses/라이센스*.txt",
+            r"licenses/ライセンス*.txt",
+            r"licenses/执照*.txt",
+            r"src/**",
+        ];
+        let cases_uv = [
+            r"public-domain/Gulliver\’s\ Travels.txt",
```

Comment:
It's verbose that you have to support all non-alphanumeric characters, but I expect that these are rare in project paths.

Reply:
I think this is the thing that perplexes me the most here. Are you doing things this way because of PEP 639? But since you're already going beyond PEP 639 by introducing escaping, does it make sense to only _require_ escaping for meta characters?